src/testdrive/src/error.rs

```diff
@@ -87,12 +107,17 @@ impl Error {
             Error::Input { details: None, .. } => {
                 panic!("programming error: print_stderr called on InputError with no details")
             }
-            Error::General { ctx, cause, hints } => {
+            Error::General { ctx, causes, hints } => {
                 let color_spec = ColorSpec::new();
                 write_error_heading(&mut stderr, &color_spec)?;
                 write!(&mut stderr, "{}", ctx)?;
-                if let Some(cause) = cause {
-                    writeln!(&mut stderr, ": {}", cause)?;
+                for (i, cause) in causes.iter().enumerate() {
+                    if i == 0 {
+                        write!(&mut stderr, ": {}", cause)?;
+                    } else {
+                        write!(&mut stderr, ", {}", cause)?;
+                    }
```

Comment:
Does this display well? I feel like it will output stuff like this that is very hard to parse when errors have nested causes:

```
error: resetting s3 state: failed to delete bucket: http connection broken, failed to delete bucket: network error: peer dropped connection, failed to delete kinesis stream: unknown stream
```

Reply:
No it's not great, but the previous implementation just threw away information. _Usually_ I would expect only one error, or all errors to have the same root cause so it's hopefully not a huge detail in the usual case.

I could switch the separator, or wrap them in parens or square brackets instead, which is uglier but less likely to be obviously confusing.

Reply:
Yeah, gotcha, it's probably fine. 🤷🏽 We can improve it if it becomes an issue!